contracts/state-validator/src/verifications/revert.rs

```diff
@@ -247,12 +248,22 @@ fn check_reverted_blocks(
     };
     let account_merkle_state = reverted_blocks[0].prev_account();
     let tip_block_hash = reverted_blocks[0].parent_block_hash();
-    let last_finalized_block_number = {
-        let number: u64 = reverted_blocks[0].number().unpack();
-        number
+    let post_version: u8 = post_global_state.version().into();
+    let last_finalized = if post_version < 2 {
+        let tip_number: u64 = reverted_blocks[0].number().unpack();
+        let finalized_number = tip_number
             .saturating_sub(1)
-            .saturating_sub(config.finality_blocks().unpack())
+            .saturating_sub(config.finality_blocks().unpack());
+        Timepoint::from_block_number(finalized_number)
+    } else {
+        let l1_timestamp = match obtain_max_timestamp_of_header_deps() {
```

Comment:
Does this mean that when reverting blocks we need to find a block whose timestamp matches the one used when submitting the last valid block? Would it be difficult?

Reply:
Whether "when reverting blocks" indicates L1 blockchain reverting or L2 blockchain reverting?

Reply:
The script currently does not enforce what the l2block.timestamp should be. So, after the L2 blockchain reverts, the timestamp of a newly created L2 block can be anything.

Reply:
Reverting as in challenge revert, i.e. submitting a `RollupRevert`.